include/miral/miral/decoration/decoration_adapter.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright © Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef MIRAL_DECORATION_DECORATION_ADAPTER_H
+#define MIRAL_DECORATION_DECORATION_ADAPTER_H
+
+#include "mir/shell/decoration_notifier.h"
```

Comment:
This should not be used by a public libmiral header.

Reply:
So, like, create a `miral::*` version of the class, with the header file completely isolating the internal symbols from the user, and the source file using/including the header? 

Does this also apply to similar cases in this review? (`input_resolver.h`, adapters, etc...)?

Reply:
No, it should just be internal and unavailable to the shell author